SignInCheckIn/SignInCheckIn/Services/ChildSigninService.cs

```diff
@@ -123,29 +123,68 @@ public ParticipantEventMapDto SigninParticipants(ParticipantEventMapDto particip
                 Contacts = participantEventMapDto.Contacts
             };
 
-            // set the data fields on the printed participant dto
-            if (eventsForSignin.Count == 2)
+            SetParticipantsPrintInformation(response.Participants, eventsForSignin);
+ 
+            // Add back those participants that didn't get a room assigned
+             response.Participants.AddRange(participantEventMapDto.Participants.Where(p => !p.AssignedRoomId.HasValue && p.Selected));
+            response.Participants.ForEach(p => p.Selected = true);
+
+            return response;
+        }
+
+        private void SetParticipantsPrintInformation(List<ParticipantDto> participants, IReadOnlyList<MpEventDto> eventsForSignin)
+        {
+            if (eventsForSignin.Count == 1)
```

Comment:
I split this into 3 function calls one that handles if count is 1 (one event) and one that handles if count is 2 (two events)